### ISSUE TITLE
[3.3.5] ObjectGuid has not been declared

### ISSUE THREAD

**Issue by Espionage724** [TrinityCore/TrinityCore#15710]
Build (and the commit I believe responsible): https://github.com/TrinityCore/TrinityCore/commit/b1032ed6203a5839906fab7137484b0ef8f0e58f

```
In file included from /home/espionage724/trinitycore/src/server/game/DataStores/DBCStores.cpp:23:0:
/home/espionage724/trinitycore/src/server/game/Maps/TransportMgr.h:114:50: error: ‘ObjectGuid’ has not been declared
         Transport* CreateTransport(uint32 entry, ObjectGuid::LowType guid = 0, Map* map = NULL);
                                                  ^
/home/espionage724/trinitycore/src/server/game/Maps/TransportMgr.h:114:70: error: expected ‘,’ or ‘...’ before ‘guid’
         Transport* CreateTransport(uint32 entry, ObjectGuid::LowType guid = 0, Map* map = NULL);
                                                                      ^
[ 25%] Building CXX object src/server/game/CMakeFiles/game.dir/DungeonFinding/LFGPlayerData.cpp.o
[ 25%] Building CXX object src/tools/mmaps_generator/CMakeFiles/mmaps_generator.dir/TerrainBuilder.cpp.o
[ 25%] Building CXX object src/server/game/CMakeFiles/game.dir/DungeonFinding/LFGQueue.cpp.o
[ 25%] Building CXX object src/server/game/CMakeFiles/game.dir/DungeonFinding/LFGScripts.cpp.o
[ 25%] Building CXX object src/tools/mmaps_generator/CMakeFiles/mmaps_generator.dir/VMapExtensions.cpp.o
[ 25%] Building CXX object src/server/game/CMakeFiles/game.dir/Entities/Corpse/Corpse.cpp.o
[ 25%] Building CXX object src/server/game/CMakeFiles/game.dir/Entities/Creature/Creature.cpp.o
[ 25%] Building CXX object src/server/game/CMakeFiles/game.dir/Entities/Creature/CreatureGroups.cpp.o
distcc[3974] ERROR: compile /home/espionage724/trinitycore/src/server/game/DataStores/DBCStores.cpp on localhost failed
src/server/game/CMakeFiles/game.dir/build.make:1358: recipe for target 'src/server/game/CMakeFiles/game.dir/DataStores/DBCStores.cpp.o' failed
make[2]: *** [src/server/game/CMakeFiles/game.dir/DataStores/DBCStores.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

I compile with:
```
CC='distcc gcc' CXX='distcc g++' cmake '/home/espionage724/trinitycore' -DTOOLS=1 -DCMAKE_INSTALL_PREFIX='/home/espionage724/run' -DWITH_WARNINGS=0 -DWITH_COREDEBUG=0 -DUSE_COREPCH=0 -DUSE_SCRIPTPCH=0 -DCMAKE_CXX_FLAGS='-O2 -pipe -march=amdfam10' -DCMAKE_C_FLAGS='-O2 -pipe -march=amdfam10'
```

Which was fine previously. I didn't see any other reports about it, so maybe it only affects non-PCH builds?

**Comment by Lillecarl** [comment]
include "ObjectGuid.h" in TransportMgr.h i guess, since you do not have PCH build enabled :)

Devs always build with PCH afaik and therefore miss things like this on occasion.
Then building with distcc makes things a bit fuzzy for me who doesn't know how it works really, but i think it sends all includes over with the source file for compilation? :)

**Comment by Espionage724** [comment]
Hmm, not entirely sure. I recall hearing in the past to disable PCH when using distcc, and I recall having some compile issue with PCH enabled with distcc when I tried it out for fun.

I'm not really certain on the specifics of it though.

**Comment by Espionage724** [comment]
Is there any special way I should be including ObjectGuid.h in TransportMgr.h? I tried just tossing it up with the rest of the includes to no avail (still get the same error above even after make clean):

```
#include "ObjectGuid.h"
#include "GameObject.h"
#include "TransportMgr.h"
#include "VehicleDefines.h"
```

**Comment by pete318** [comment]
I'll check with PCH off and fix after work. In the meantime maybe add the include as suggested.

Interesting that Travis also builds this way.

**Comment by jackpoz** [comment]
Travis builds with https://travis-ci.org/TrinityCore/TrinityCore/builds/85003802#L297
>cmake ../ -DWITH_WARNINGS=1 -DWITH_COREDEBUG=0 -DUSE_COREPCH=1 -DUSE_SCRIPTPCH=1 -DTOOLS=1 -DSCRIPTS=1 -DSERVERS=1 -DNOJEM=1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS="-Werror" -DCMAKE_CXX_FLAGS="-Werror"

so it's never going to report broken no-pch builds

**Comment by pete318** [comment]
Should now be fixed with 3f29ce9e8f7c9c57eacf38493737b5a7bce39925
